compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -5260,6 +5260,111 @@ static bool skipRemLowering(int64_t divisor, TR::Simplifier *s)
    }
 
 
+/* Reduce Ifcmp Greater-Than-Or-Equal to Equal in the following case. If
+* - the first child is an "and" node
+* - the second child of the "and" node is a constant N
+* - the second child of the ifcmple node is a constant N,
+* Then, we can reduce ifcmple to ifcmpeq.
+*
+* For example, convert
+* ifiucmple --> block_5
+*  iand
+*    iload  <...>
+*    iconst 16
+*  iconst 16
+*
+*  into
+*
+* ifiucmpeq --> block_5
+*  iand
+*    iload  <...>
+*    iconst 16
+*  iconst 16
+*
+* Since the constants are equal, the case of greater-than/less-than is impossible,
+* and we can reduce a GE/LE operation to an EQ operation. After the reduction is performed,
+* the test-under-mask optimization can be applied to the "ifcmpeq" node.
+*/
+class IfcmpReducer
+   {
+   public:
+   IfcmpReducer(TR::Node *node, TR::Simplifier *s) : _node(node), _s(s) {}
+
+   static TR::ILOpCodes getNewOpCode(TR::Node *node)
+      {
+      auto opCodeValue = node->getOpCodeValue();
+      if (opCodeValue == TR::ifbcmple || opCodeValue == TR::ifbcmpge)
```

Comment:
The `if` `else` seems overkill for this many cases. Is it possible to use a `switch` here?